nixos/modules/services/mail/rspamd.nix

```diff
@@ -294,20 +319,44 @@ in
       gid = config.ids.gids.rspamd;
     };
 
-    environment.etc."rspamd.conf".source = rspamdConfFile;
+    services.rspamd = {
+      socketActivation = mkDefault (!opts.bindSocket.isDefined && !opts.bindUISocket.isDefined);
+
+      workers = mkIf cfg.postfix.enable {
+        controller = {};
+        rspamd_proxy = {
+          bindSockets = [ {
+            socket = "/var/lib/postfix/queue/private/rspamd";
+            owner = "rspamd";
+            group = "postfix";
+          } ];
+        };
+      };
+
+      overrides."logging.inc" = mkDefault ''
+        type = "syslog";
+      '';
+    };
+
+    services.postfix.extraConfig = mkIf cfg.postfix.enable ''
+      smtpd_milters = unix:private/rspamd
+      non_smtpd_milters = $smtpd_milters
+      milter_protocol = 6
+    '';
+
+    environment.etc."rspamd".source = rspamdConf;
 
     systemd.services.rspamd = {
       description = "Rspamd Service";
 
       wantedBy = mkIf (!cfg.socketActivation) [ "multi-user.target" ];
-      after = [ "network.target" ] ++
-       (if cfg.socketActivation then allSocketNames else []);
-      requires = mkIf cfg.socketActivation allSocketNames;
```

Comment:
Why remove these? In your commit message you mention that service dependencies where changed to counteract the sockets getting deleted but that problem is fixed further down by not setting `RuntimeDirectory`